central/metadata/service/service_impl.go

```diff
@@ -48,6 +51,14 @@ func (s *serviceImpl) GetMetadata(ctx context.Context, _ *v1.Empty) (*v1.Metadat
 		ReleaseBuild:  buildinfo.ReleaseBuild,
 		LicenseStatus: v1.Metadata_VALID,
 	}
+	if marketing.Enabled() {
+		config, err := mpkg.GetDeviceConfig()
+		if err == nil {
+			metadata.CentralId = config.ID
+			metadata.OrganizationId = config.OrgID
+			metadata.SegmentKey = env.SegmentAPIKey.Setting()
+		}
```

Comment:
```suggestion
		if config, err := mpkg.GetDeviceConfig(); err == nil {
			metadata.CentralId = config.ID
			metadata.OrganizationId = config.OrgID
			metadata.SegmentKey = env.SegmentAPIKey.Setting()
		}
```
Should we log the error here?

Reply:
Today, `/v1/metadata` is called by the frontend periodically for a connection check. We don't want to log an error here, as it would probably be repetitive. The error will be printed by another user  of `GetDeviceConfig()`.
I noticed, though, that `config` could be `nil`, so I'll adapt the condition to check `config` instead.